website/docs/reference/widgets/table/inline-editing.md

```diff
@@ -235,13 +235,117 @@ This property contains the details of the row that triggered the action (`onSubm
+
+The subheadings below describe these properties in more detail:
+
+#### Allow adding a row
+
+Turning this on allows users to click a button on the table to start a new row in the table. Users can only input data for columns that are marked as **Editable** in the table properties.
+
+Use the **onSave** event to update the source of the table's data and reflect the user's changes. For example, the **onSave** event might be set to trigger a query that sends the data from a new table row to the database that supplies the table with its values.
+
+#### Default Values
+
+<VideoEmbed host="youtube" videoId="3KozovP7LNk" title="Table | Add new rows | Default values" caption="Set default values for new Table rows"/>
+
+The values to automatically populate the new row with when a user begins creating a new row. Expects an object with the same keys as the columns in the existing table data.
+
+#### isAddRowInProgress
+
+While the "Add new row" mode is enabled in the Table widget, `isAddRowInProgress` is `true`. You can use this for a variety of reasons, such as providing visual feedback in the UI for the user or for enforcing specific validation rules for user input in new rows (See [isNewRow](#isnewrow)).
+
+#### newRow
+
+When a new row is being added to the Table, this variable contains a reference to that new row object being added. When there is not a new row in progress, `newRow` is `undefined`.
+
+This is useful for accessing the user input to send it via query once the user is ready to submit the new data.
+
+#### isNewRow
+
+`isNewRow` can be used only within the **Valid** property. With it, you may set specific validation rules for new table rows that does not neccesarily apply to editing existing cells.
+
+In the following video example, the Table allows editing a client in "Zone AE," however it does not allow adding any _new_ rows in Zone AE.
+
+<VideoEmbed host="youtube" videoId="d8hpCBW6rGI" title="Table | Add new rows | isNewRow Validation" caption="Set specific validation rules for new rows"/>
+
+```javascript
+// in the Table widget's Valid property:
+{{isNewRow? Table1.currentRow.zone !== "AE" : true}}
+```
+
+A user may make any changes to existing rows, however they are not allowed to add any new row in AE.
+
+### Events
+
+| Events        | Description                                                                                                            |
+| ------------- | ---------------------------------------------------------------------------------------------------------------------- |
+| **onSave**    | Action that gets triggered when the user clicks the save button for a cell or a new row.                               |
+| **onDiscard** | Action that gets triggered when user clicks discard button for a cell or a new row.                                    |
 
 ## What's next?
```

Comment:
The What's next section here seems inappropriate. I know it was there before. Maybe we should revisit this as part of rehaul.  It should be either something like tutorials or articles , common errors, troubleshooting, sample apps all related to this topic on inline editing.

Reply:
I created a ticket for rehauling this page, which i will pick up after this PR; included this point and some other action items

Reply:
#18170